va/va.go

```diff
@@ -254,38 +254,39 @@ type verificationRequestEvent struct {
 // passing through the detailed message.
 func detailedError(err error) *probs.ProblemDetails {
 	// net/http wraps net.OpError in a url.Error. Unwrap them.
-	if urlErr, ok := err.(*url.Error); ok {
+	var urlErr *url.Error
+	if errors.As(err, &urlErr) {
 		prob := detailedError(urlErr.Err)
 		prob.Detail = fmt.Sprintf("Fetching %s: %s", urlErr.URL, prob.Detail)
 		return prob
 	}
 
-	if tlsErr, ok := err.(tls.RecordHeaderError); ok && bytes.Compare(tlsErr.RecordHeader[:], badTLSHeader) == 0 {
+	var tlsErr tls.RecordHeaderError
+	if errors.As(err, &tlsErr) && bytes.Compare(tlsErr.RecordHeader[:], badTLSHeader) == 0 {
 		return probs.Malformed("Server only speaks HTTP, not TLS")
 	}
 
 	var netErr *net.OpError
+	var syscallErr *os.SyscallError
 	if errors.As(err, &netErr) {
 		if fmt.Sprintf("%T", netErr.Err) == "tls.alert" {
 			// All the tls.alert error strings are reasonable to hand back to a
 			// user. Confirmed against Go 1.8.
 			return probs.TLSError(netErr.Error())
-		} else if syscallErr, ok := netErr.Err.(*os.SyscallError); ok &&
-			syscallErr.Err == syscall.ECONNREFUSED {
+		} else if errors.As(err, &syscallErr) && syscallErr.Err == syscall.ECONNREFUSED {
```

Comment:
Looking at this code again, I'd like to improve its organization as part of this change (since this change makes the disorganization more obvious).

Specifically, I think we should have three `if` blocks, each with a `var fooError foo.Error` above it:

```
var opErr *net.OpError
if errors.As(err, &opErr) {
 .. netErr conditions that early-return
}

var syscallErr *os.SyscallError
if errors.As(err, &syscallErr) {
 ...
}

var netErr net.Error
if errors.As(err, &netErr) {
...
}
```

(note that we currently have a variable called `netErr` which is an `OpError`; we should fix that).

Also, the branches for syscallErr should be a switch, I think.